src/main/java/gov/usgs/owi/nldi/controllers/LinkedDataController.java

```diff
@@ -199,18 +200,19 @@ public void getFlowlines(HttpServletRequest request, HttpServletResponse respons
 			@PathVariable(Parameters.FEATURE_ID) String featureID,
 			@PathVariable(Parameters.NAVIGATION_MODE) @Pattern(regexp=REGEX_NAVIGATION_MODE) String navigationMode,
 			@RequestParam(value=Parameters.STOP_COMID, required=false) @Range(min=1, max=Integer.MAX_VALUE) String stopComid,
-			@ApiParam(value=Parameters.DISTANCE_DESCRIPTION)
-								 @RequestParam(value=Parameters.DISTANCE, required=false)
-								 @Range(min=1, max=9999, message="distance must be between 1 and 9999 kilometers") String distance,
-			@RequestParam(value=Parameters.LEGACY, required=false) String legacy) {
+			@ApiParam(value=Parameters.DISTANCE_DESCRIPTION) @RequestParam(value=Parameters.DISTANCE, required=false)
+			@Pattern(message=Parameters.DISTANCE_VALIDATION_MESSAGE, regexp=Parameters.DISTANCE_VALIDATION_REGEX) String distance,
+			@RequestParam(value=Parameters.LEGACY, required=false) String legacy) throws Exception {
 
 		BigInteger logId = logService.logRequest(request);
+
 		try {
+			String guaranteedDistance = (StringUtils.isEmpty(distance)) ? Parameters.MAX_DISTANCE: distance;
```

Comment:
Spring's RequestParam accepts a defaultValue which may accomplish the defaulting of the value.